template.yaml

```diff
@@ -4033,6 +4036,36 @@ Resources:
                 authenticationCallbackUserinfoKeyArn,
               ]
 
+  RpPublicKeyCacheTableReadAccessPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Sid: AllowRpPublicKeyCacheTableReadAccess
+            Effect: Allow
+            Action:
+              - dynamodb:DescribeTable
+              - dynamodb:Get*
+            Resource: !GetAtt RpPublicKeyCacheTable.Arn
+          - Sid: AllowRpPublicKeyCacheTableKeyAccess
+            Effect: Allow
+            Action:
+              - kms:Decrypt
+            Resource: !GetAtt RpPublicKeyTableEncryptionKey.Arn
+
+  RpPublicKeyCacheTableWriteAccessPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Sid: AllowRpPublicKeyCacheTableWriteAccess
```

Comment:
does this need something like a kms generate data key?

Reply:
In order to encrypt the table? In my testing this hasn't seemed necessary

Reply:
Will bear it in mind in case it comes up later on